lib/collector/api.js

```diff
@@ -117,6 +117,22 @@ CollectorAPI.prototype.connect = function connect(callback) {
     this._reqHeadersMap = null
   }
 
+  /**
+   * Checks if proxy is configured to connect via `proxy_host` and `proxy_port`
+   * and if error code is EPROTO or ECONNRESET. This is an indication their proxy
+   * server only accepts HTTP connections, and we should provide an actionable warning to
+   * fix the misconfiguration by setting `proxy` to a fully qualified URL
+   *
+   * @param {Error} error
+   * @returns {Boolean}
+   */
+  function isProxyMisconfigured(error) {
+    const config = api._agent.config || {}
```

Comment:
this might be super defensive

Reply:
Yeah, I think we always get at least the default config here.